types/global.d.ts

```diff
@@ -191,7 +196,7 @@ declare global {
     getContractFactory<
       TContractFactory extends ContractFactory = ContractFactory
     >(
-      name: ContractNames,
+      name: string,
```

Comment:
I can see how this generic can be frustrating to work with. I wrote this so to add type support and intellisense support when writing contracts deployments.


It also once helped with type inference when deploying contracts so you don't have to do `as ContractName` but that part appears to have broke (for some reason ContractFactory is actually causing a type error because it's not actually imported in global.d.ts and I think that's causing ts to infer a type of `any`)

Did you run  into issues with this? If it can't be reverted back to `ContractNames`, we should just delete this and stop trying to augment the types. 

(it's also possible the type augmentation i added here isn't required anymore and inference might be working by default in the latest types of various packages)

Reply:
It had some trickle down effects with the renaming if I recall correctly.  Including `LockedNORI` as well as `LockedNORIV2` here causes it to clash with several other types that I'd like top leave `LockedNORI` out of.  Similar story for any solidity test contracts.  I will bookmark this to revisit when the dust settles as well.